Sources/LogtoClient/LogtoClient/LogtoClient+SignIn.swift

```diff
@@ -32,9 +33,43 @@ public extension LogtoClient {
         )
     }
 
+    // TO-DO: error handling
+    // TO-DO: implement full functions
     func signInWithBrowser() {
-        fetchOidcConfig { _, _ in
-            // To implement
+        fetchOidcConfig { [self] oidcConfig, _ in
+            guard let oidcConfig = oidcConfig else {
+                return
+            }
+
+            let scheme = "io.logto.SwiftUI-Demo"
+            let state = LogtoUtilities.generateState()
+            let codeVerifier = LogtoUtilities.generateCodeVerifier()
+            let codeChallenge = LogtoUtilities.generateCodeChallenge(codeVerifier: codeVerifier)
+
+            guard let authUri = try? LogtoCore.generateSignInUri(
+                authorizationEndpoint: oidcConfig.authorizationEndpoint,
+                clientId: logtoConfig.clientId,
+                redirectUri: "\(scheme)://callback",
+                codeChallenge: codeChallenge,
+                state: state,
+                scope: nil,
+                resource: nil
```

Comment:
I think we just aligned to use `scopes` and `resources`

Reply:
yes, will do in a separate PR